pkg/apis/core/validation/validation_test.go

```diff
@@ -6362,80 +6362,97 @@ func TestValidateMountPropagation(t *testing.T) {
-		defaultContainer,
-		true,
+		mount:       core.VolumeMount{Name: "foo", MountPath: "/foo", MountPropagation: &propagationBidirectional},
+		container:   defaultContainer,
+		expectError: true,
 	}, {
 		// explicitly non-privileged container + no propagation
-		core.VolumeMount{Name: "foo", MountPath: "/foo"},
-		nonPrivilegedContainer,
-		false,
+		mount:       core.VolumeMount{Name: "foo", MountPath: "/foo"},
+		container:   nonPrivilegedContainer,
+		expectError: false,
 	}, {
 		// explicitly non-privileged container + HostToContainer
-		core.VolumeMount{Name: "foo", MountPath: "/foo", MountPropagation: &propagationHostToContainer},
-		nonPrivilegedContainer,
-		false,
+		mount:       core.VolumeMount{Name: "foo", MountPath: "/foo", MountPropagation: &propagationHostToContainer},
+		container:   nonPrivilegedContainer,
+		expectError: false,
 	}, {
 		// explicitly non-privileged container + HostToContainer
-		core.VolumeMount{Name: "foo", MountPath: "/foo", MountPropagation: &propagationBidirectional},
-		nonPrivilegedContainer,
-		true,
+		mount:       core.VolumeMount{Name: "foo", MountPath: "/foo", MountPropagation: &propagationBidirectional},
+		container:   nonPrivilegedContainer,
+		expectError: true,
 	}, {
 		// privileged container + no propagation
-		core.VolumeMount{Name: "foo", MountPath: "/foo"},
-		privilegedContainer,
-		false,
+		mount:     core.VolumeMount{Name: "foo", MountPath: "/foo"},
+		container: privilegedContainer,
 	}, {
 		// privileged container + HostToContainer
-		core.VolumeMount{Name: "foo", MountPath: "/foo", MountPropagation: &propagationHostToContainer},
-		privilegedContainer,
-		false,
+		mount:       core.VolumeMount{Name: "foo", MountPath: "/foo", MountPropagation: &propagationHostToContainer},
+		container:   privilegedContainer,
+		expectError: false,
 	}, {
 		// privileged container + Bidirectional
-		core.VolumeMount{Name: "foo", MountPath: "/foo", MountPropagation: &propagationBidirectional},
-		privilegedContainer,
-		false,
+		mount:       core.VolumeMount{Name: "foo", MountPath: "/foo", MountPropagation: &propagationBidirectional},
+		container:   privilegedContainer,
+		expectError: false,
 	}, {
 		// error: privileged container + invalid mount propagation
-		core.VolumeMount{Name: "foo", MountPath: "/foo", MountPropagation: &propagationInvalid},
-		privilegedContainer,
-		true,
+		mount:       core.VolumeMount{Name: "foo", MountPath: "/foo", MountPropagation: &propagationInvalid},
+		container:   privilegedContainer,
+		expectError: true,
+	}, {
+		// explicitly non-privileged container with required capability + Bidirectional
+		mount:     core.VolumeMount{Name: "foo", MountPath: "/foo", MountPropagation: &propagationBidirectional},
+		container: nonPrivilegedContainerWithCapability,
+		opts: PodValidationOptions{
+			AllowUnprivilegedBidirectionalMountPropagation: true,
+		},
+		expectError: false,
```

Comment:
Please add a test with `nonPrivilegedContainerWithCapability` and `AllowUnprivilegedBidirectionalMountPropagation: false` (or not set), it should throw an error.